gulpfile.js

```diff
@@ -0,0 +1,105 @@
+
+var gulp = require('gulp'),
+  runSequence = require('run-sequence'),
+  del = require('del'),
+  replace = require('gulp-string-replace'),
+  sourcemaps = require('gulp-sourcemaps'),
+  exec = require('child_process').exec,
+  ngc = require('gulp-ngc'),
+  changed = require('gulp-changed');
+
+var appSrc = 'src';
+var libraryDist = 'dist';
+var watchDist = 'dist-watch';
+var globalExcludes = [ '!./**/examples/**', '!./**/examples' ]
+
+/**
+ * FUNCTION LIBRARY
+ */
+
+function copyToDist(srcArr) {
+  return gulp.src(srcArr.concat(globalExcludes))
+    .pipe(gulp.dest(function (file) {
+      return libraryDist + file.base.slice(__dirname.length); // save directly to dist
+    }));
+}
+
+function updateWatchDist() {
+  return gulp
+    .src([libraryDist + '/**'].concat(globalExcludes))
+    .pipe(changed(watchDist))
+    .pipe(gulp.dest(watchDist));
+}
+
+/**
+ * TASKS
+ */
+
+gulp.task('post-transpile', ['transpile'], function () {
+  return gulp.src(['dist/src/app/**/*.js'])
+    .pipe(replace(/templateUrl:\s/g, "template: require("))
+    .pipe(replace(/\.html',/g, ".html'),"))
+    .pipe(replace(/\.html'/g, ".html')"))
+    .pipe(replace(/styleUrls: \[/g, "styles: [require("))
+    .pipe(replace(/\.less']/g, ".css').toString()]"))
+    .pipe(gulp.dest(function (file) {
+      return file.base; // because of Angular 2's encapsulation, it's natural to save the css where the less file was
+    }));
+});
+
+//Less compilation and minifiction - adopted from sass compilation, needs work
+gulp.task('transpile-less', function () {
+  return transpileLESS(appSrc + '/app/**/*.less');
+});
+
+// Put the SASS files back to normal
```

Comment:
It is only going to be LESS - there is another issue to manage CSS (which this does not cover).  @mindreeper2420